SmartDeviceLink/SDLChoiceSetManager.m

```diff
@@ -288,16 +324,27 @@ - (void)deleteChoices:(NSArray<SDLChoiceCell *> *)choices {
     [self sdl_findIdsOnChoices:cellsToBeDeleted];
     SDLDeleteChoicesOperation *deleteOp = [[SDLDeleteChoicesOperation alloc] initWithConnectionManager:self.connectionManager cellsToDelete:cellsToBeDeleted];
 
-    __weak typeof(self) weakself = self;
+    __weak typeof(self) weakSelf = self;
     __weak typeof(deleteOp) weakOp = deleteOp;
     deleteOp.completionBlock = ^{
+        SDLLogD(@"Finished deleting choices");
+
+        __strong typeof(weakSelf) strongSelf = weakSelf;
         if (weakOp.error != nil) {
             SDLLogE(@"Failed to delete choices: %@", weakOp.error);
             return;
         }
 
-        SDLLogD(@"Finished deleting choices");
-        [weakself.preloadedMutableChoices minusSet:cellsToBeDeleted];
+        // Check if the manager has shutdown because the list of uploaded choices should not be updated
+        if ([strongSelf.currentState isEqualToString:SDLChoiceManagerStateShutdown]) {
```

Comment:
Instead of checking the state here and in other blocks, we could just "reset" all the properties in the `start` method.

Reply:
I'm not sure what you mean by that. Why would resetting properties in `start` help? They're already reset in `didEnterStateShutdown`.

Reply:
The problem is that the `didEnterStateShutdown` is called and all the properties reset before the completion block is called for the choice set upload. Then in the completion block, some of the properties are set.

Reply:
Ah I gotcha. I think this is a better way to do it. Better not to set the properties at all than to have to reset them later. If a dev checks them while it's shut down it'll show the wrong data.